h/groups/views.py

```diff
@@ -101,7 +105,28 @@ def _read_group(request, group):
 
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
-    template_data = {'group': group, 'group_url': url}
+
+    result = search.search(request, private=False, params={
+        "group": group.pubid, "limit": 1000})
+    annotations = result['rows']
+
+    # Group the annotations by URI.
+    # Create a dict mapping the (normalized) URIs of the annotated documents
+    # to the most recent annotation of each document.
+    annotations_by_uri = collections.OrderedDict()
```

Comment:
I'm not sure if you'd say this is easier or harder to read, but it makes a bit more sense to me:
```
buckets = itertools.groupby(annotations, lambda a: uri.normalize(a.uri))
document_links = itertools.islice(next(anns).document_link for _, anns in buckets), 25)
```